driver-core/src/main/com/mongodb/internal/async/function/RetryState.java

```diff
@@ -182,6 +182,9 @@ private void doAdvanceOrThrow(final Throwable attemptException,
             final boolean onlyRuntimeExceptions) throws Throwable {
         assertTrue(attempt() < attempts);
         assertNotNull(attemptException);
+        if (attemptException instanceof MongoOperationTimeoutException) {
```

Comment:
This conditional seems unrelated to the rest of the PR. What's the intention of including it?

Reply:
Had a test failing due to the wrapping of the exception that later occurs.